packages/truffle-core/lib/command.js

```diff
@@ -100,9 +100,39 @@ class Command {
       }
     });
 
-    const newOptions = Object.assign({}, clone, argv);
-
+    //Check unsupported command line flag according to the option list in help
     try {
+      const inputOptions = [];
+      inputStrings.map(i => {
```

Comment:
Instead of creating an array first and then using the `push` method, just set `inputStrings` like the following:
```
const inputOptions = inputStrings.map(string => {
  return string.startsWith("--") ? string : null;
}).filter(item => item);
```
`map` already returns an array but you also need to filter out the `null`s afterwards.